### PR TITLE
Improve git icons on directories

### PR DESCRIPTION
**Pull request by chomosuke**
After adding option git.show_on_open_dirs, I've discovered some undesirable behavior:
- ignored directory's ignored status going away after opening it.
- having no way of knowing a file a deleted once the directory it's in is opened.

This PR fixes the above two short coming and also made parent directories show all the git status icons from their children.

The ordering of the git icons in `lua/nvim-tree/renderer/components/git.lua:10` are made to preserve the original ordering of some of the multi-icons status. I'd be happy to change it if anyone have different preferences :).

### CODE REVIEW

**Comment by alex-courtis** [comment]
> The ordering of the git icons in `lua/nvim-tree/renderer/components/git.lua:10` are made to preserve the original ordering of some of the multi-icons status. I'd be happy to change it if anyone have different preferences :).

I think we can remove that. The order of statuses to icons retrieved by `git.get_icons` is Good Enough and easy to change following its normalisation.

**Review by alex-courtis — CHANGES_REQUESTED** [review]
I've tested a variety of cases, however it's not practical to test all cases.

Please:

- [ ] use natural order for statuses: https://github.com/nvim-tree/nvim-tree.lua/pull/1809#issuecomment-1345438281
- [ ] `common.lua` -> `node.lua`
- [ ] use `git_statuses` `git_status` consistently in all of `node.lua`
- [ ] merge master and retest: #1784 introduced some changes to git statuses

Thank you so much: this refactor and consolidation is a great improvement to the codebase.

**Comment by chomosuke** [comment]
> > The ordering of the git icons in `lua/nvim-tree/renderer/components/git.lua:10` are made to preserve the original ordering of some of the multi-icons status. I'd be happy to change it if anyone have different preferences :).
> 
> I think we can remove that. The order of statuses to icons retrieved by `git.get_icons` is Good Enough and easy to change following its normalisation.

Hmmm are you sure? Imo it looks quite weird having folders with the same combination of changes being different, for example:

image

Here both marks and renderer have some files changed and some new files. But it looks like they have different git status if that make sense.

Though tbh even with their icons sorted they'd still have different highlight. Maybe what we need is to sort the statuses?

**Comment by alex-courtis** [comment]
> Here both marks and renderer have some files changed and some new files, but it looks like they have different git status.
> 
> Though tbh even with their icons sorted they'd still have different highlight. Maybe what we need is to sort the statuses?

Whoops... I didn't notice that it was a maplike table. Options:

1. revert 23f6276ef76ccb12e8c9e02550d33525c469154a
2. change it to an array-like with `staged` etc. as a table member and use `utils.key_by`

I like 1.

File: lua/nvim-tree/renderer/components/git.lua [hunk 1]
```diff
@@ -60,19 +60,19 @@ local function warn_status(git_status)
 end
 
 local function get_icons_(node)
-  local git_statuses = explorer_common.get_git_status(node)
-  if git_statuses == nil then
+  local git_status = explorer_common.get_git_status(node)
```
Comment (alex-courtis) [inline lua/nvim-tree/renderer/components/git.lua]: I was actually thinking we could use `git_statuses` everwhere as it is a much better variable name, similar to `iconss`.

However... looking through the whole codebase we are using `git_status` _everywhere_ so I guess that change is outside of the scope of this PR.

**Review by alex-courtis — CHANGES_REQUESTED** [review]
Please undo my bad idea: https://github.com/nvim-tree/nvim-tree.lua/pull/1809#issuecomment-1354132930

File: lua/nvim-tree/renderer/components/git.lua (hunk 1 above)
Comment (chomosuke) [reply lua/nvim-tree/renderer/components/git.lua]: Ahhh sorry, I misunderstood. Perhaps I'll open a PR right after this one specifically for git_statuses?

Comment (alex-courtis) [reply lua/nvim-tree/renderer/components/git.lua]: If you're keen I would be most grateful! There are a lot of usages...

**Review by alex-courtis — APPROVED** [review]

**Comment by alex-courtis** [comment]
Many thanks for your contribution
